### PR TITLE
feat: list view counts

### PR DESCRIPTION
**Pull request by davidruvolo51**
This PR is a continuation of #3469 which added a component to display the number of search results in each view. Search results are now displayed in each view (cohorts, data sources, and variables) and e2e testing is available for all pages.

how to test:
- Navigate to the preview:
- Navigate to the Cohorts, Data Sources, and Variables pages
- Note the counts above the results
- Apply filters and note the change

todo:
- [x] added tests

### CODE REVIEW

**Comment by davidruvolo51** [comment]
Part of issue #3475

**Comment by davidruvolo51** [comment]
I had some issues running the e2e test locally. Tests would fail when the following line was enabled

```js
await page.getByRole('button', { name: 'Reject' }).click();
```

Perhaps we can rewrite this step so that it skips the test if running locally?

**Comment by connoratrug** [comment]
We might even hide the count as there is a message shown
Screenshot 2024-03-08 at 13 57 35

**Review by connoratrug — APPROVED** [review]
works as expected, would suggest to always use a number as a return type for the numberOfCohorts ( see comment)

File: e2e/tests/catalogue/search-results-count.spec.ts [hunk 1]
```diff
@@ -1,7 +1,31 @@
 import { test, expect } from '@playwright/test';
 
-test('test', async ({ page }) => {
-  await page.goto('/catalogue-demo/ssr-catalogue/all/cohorts');
-  await page.getByRole('button', { name: 'Reject' }).click();
-  await expect(page.getByRole('main')).toContainText('53 cohorts');
-});
\ No newline at end of file
+const pattern = new RegExp(/^(([a-zA-Z]{1,})?(\s)?(([0-9]{1,})\s(cohort([s])?|variable([s])?|data\ssource([s])?|result([s])?)))$/);
```
Comment (connoratrug) [inline e2e/tests/catalogue/search-results-count.spec.ts:3]: o probably, good to name the pattern , give it a descriptive name

**Comment by davidruvolo51** [comment]
I will create a new issue for handling 0 results. Currently, if there are no results, the content shifts when the "No records found" message is displayed.. Perhaps there is some box-sizing or padding around text elements that's causing the shift.

**Pull request merged**